src/commands/service/create.ts

```diff
@@ -10,328 +10,144 @@ import {
   addNewService as addNewServiceToBedrockFile,
   YAML_NAME as BedrockFileName
 } from "../../lib/bedrockYaml";
+import { build as buildCmd, exit as exitCmd } from "../../lib/commandBuilder";
 import {
   addNewServiceToMaintainersFile,
   generateDockerfile,
   generateGitIgnoreFile,
   generateStarterAzurePipelinesYaml
 } from "../../lib/fileutils";
 import { checkoutCommitPushCreatePRLink } from "../../lib/gitutils";
+import { isPortNumber } from "../../lib/validator";
 import { logger } from "../../logger";
 import { IBedrockFileInfo, IHelmConfig, IUser } from "../../types";
+import decorator from "./create.decorator.json";
+
+export interface ICommandOptions {
+  displayName: string;
+  gitPush: boolean;
+  helmChartChart: string;
+  helmChartRepository: string;
+  helmConfigBranch: string;
+  helmConfigGit: string;
+  helmConfigPath: string;
+  k8sBackend: string;
+  maintainerEmail: string;
+  maintainerName: string;
+  middlewares: string;
+  packagesDir: string;
+  pathPrefix: string;
+  pathPrefixMajorVersion: string;
+  k8sBackendPort: string;
+}
+
+export interface ICommandValues extends ICommandOptions {
+  k8sPort: number;
+  middlewaresArray: string[];
+  variableGroups: string[];
+}
+
+export const fetchValues = (opts: ICommandOptions) => {
+  if (!isPortNumber(opts.k8sBackendPort)) {
+    throw new Error("value for --k8s-service-port is not a value port number");
+  }
+
+  let variableGroups: string[] = [];
+
+  try {
+    const bedrock = Bedrock();
+    variableGroups = bedrock.variableGroups || [];
+  } catch (_) {
+    // NO-OP
```

Comment:
why is this a no-op?

Reply:
sorry, I should have added some comments. For testing (running the command), I found that this command is ok without bedrock.yaml. the presence of this yaml file is to get the variable groups information, please let me know if this not accurate. That's bedrock.yaml file is needed and I will amend the changes. Otherwise I will add some comments like this 
```
bedrock.yaml file is optional. We get variable groups information in it if it exists
```

Reply:
May be we need to extend test cases. We check if the user ran `project init` and `project cvg` commands before running this command so `bedrock.yaml` should exist with at least one variable group. The below line performs the validation.

https://github.com/CatalystCode/spk/blob/d1e0ac09ecb9e74808c02d57536facef640b20c3/src/commands/service/create.ts#L110

Reply:
can you please create a task? And we can do this in a separate PR

Reply:
That will be fine. I think we can remove try/catch block here since the code never reaches here without a `bedrock.yaml` file in the directory.

Reply:
ok. I will remove the try-catch block. That's this command needs bedrock.yaml otherwise it will fail. thanks